Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/NamespaceSynchronizationConfig.swift

```diff
@@ -155,17 +156,17 @@ internal class NamespaceSynchronization: Sequence {
     subscript(documentId: BSONValue) -> CoreDocumentSynchronization? {
         get {
             nsLock.readLock()
-            defer { nsLock.unlock() }
+            defer { nsLock.unlock(for: .reading) }
             guard var config = config.syncedDocuments[HashableBSONValue(documentId)] else {
                 return nil
             }
-            return CoreDocumentSynchronization.init(docsColl: docsColl,
-                                                    config: &config,
-                                                    errorListener: errorListener)
+            return try? CoreDocumentSynchronization.init(docsColl: docsColl,
```

Comment:
[for further discussion] I don't think we should be squashing errors like this

Reply:
Will do in the error ticket.